### DISCUSSION TITLE
Automatic session encryption key handling

### DISCUSSION THREAD

**Discussion by mrx23dot** [element-hq/element-meta#969] [Ideas]
#### What would you like to do?
Can we make session key backup/restore automatic/seamless?

#### Why would you like to do it?
It's so annoying that I have to micromanage the encryption itself, we all got better things to do in life.
Not whatsapp/signal/wire require us to do the same for e2ee, they just work out of the box.

It's totally fine not to see history before the point I joined a room, but **I shouldn't need to do anything to see everything after**.

#### How would you like to achieve it?
You tell me how signal achieves this (as I know it's built onto the same protocol), but I guess auto save/restore keys on login, and not asking anything from the user, but for tech savvy people you can have a hidden option to save/load keys from file.

Ask yourself would your mum be able to use the system? (does it work out of the box?)

### Have you considered any alternatives?

whatsapp/signal/wire work out of the box when e2ee enabled, no need to do anything extra.

**Comment by eNkKVS4HxHc9jaq** [comment]
A little offtopic here but we can't even be sure whether WhatsApp has E2EE. They claim to but truly they don't and known backdoors have been discovered in the past.
